client/serialize/rust/src/serializer/mod.rs

```diff
@@ -0,0 +1,112 @@
+use eyre::{eyre, Result, WrapErr};
+use flatbuffers::{FlatBufferBuilder, WIPOffset};
+use tp_client::contract::properties::states::IStates;
+
+use crate::baseline::BaselineArgs;
+use crate::contract::{ContractArgs, ContractIdArgs, ContractStatesArgs};
+use crate::{c, t};
+
+pub struct Serializer<'b> {
+    fbb: FlatBufferBuilder<'static>,
+    baseline: &'b c::Baseline,
+    contracts: Vec<WIPOffset<t::Contract<'static>>>,
+    states: Vec<WIPOffset<t::State<'static>>>,
+    objects: Vec<WIPOffset<t::Object<'static>>>,
```

Comment:
Are these `WIP` types to be replaced in this PR's parent task?

Reply:
These types are from flatbuffer, and represent handles to tables inside the unfinished flatbuffer.

They get used to create a vector, in the `finish` method.